test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
                          :else (do
                                  (Thread/sleep 500)
                                  (recur (dec times)))))))))))
+
+
+
+(deftest ^:integration revoke-compile-master-test
+  (testing "Revoking compile master certificate"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
+        (testing "should update infrastructure CRL"
+          (let [ca-cert (bootstrap/get-ca-cert-for-running-server)
+                subject "dummy-compile-master"
+                ;; In one of the other unit tests I noticed the localhost cert is
+                ;; being revoked. If that does not affect things we could do that instead of 
+                ;; creating a new one. But wondering it actually may have impact on other 
+                ;; tests since this is integration
```

Comment:
The `bootstrap/with-puppetserver-running-with-mock-jrubies` will spin up a new instance of puppetserver, you can give it a custom confdir in your configuration which will move the default cadir as well. You can create fixtures in that cadir that will not affect other tests. See [this CA test](https://github.com/puppetlabs/puppetserver/blob/master/test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj#L185-L203) for an example of setting the confdir to [this test fixture directory](https://github.com/puppetlabs/puppetserver/tree/master/dev-resources/puppetlabs/services/certificate_authority/certificate_authority_int_test/ca_true_test/master/conf) that the test then manipulates the contents of the csr directory.

Reply:
I think I will just issue a new cert and so should not have to move the configuration. A new cert and infra CRL will not affect anything.